SampleApp/Program.cs

```diff
@@ -0,0 +1,95 @@
+        /// Session key for the cached forecast data
+        /// </summary>
+        private const string ForecastCacheKey = "forecast";
+
+        public static void Main(string[] args)
+        {
+            var builder = WebApplication.CreateBuilder(args);
+
+            // Add services to the container.
+            builder.Services.AddAuthorization();
+
+            // Add session state support
+            builder.Services.AddSession(options =>
+            {
+                options.IdleTimeout = TimeSpan.FromSeconds(30);
+                options.Cookie.IsEssential = true;
+            });
+
+            // Add DynamoDB distributed cache for sessions
+            builder.Services.AddAWSDynamoDBDistributedCache(options =>
+            {
+                options.TableName = "weather-sessions-cache";
+                options.CreateTableIfNotExists = true;
+                options.UseConsistentReads = true;
+                options.PartitionKeyName = "id";
+                options.TTLAttributeName = "cache_ttl";
+            });
+
+            // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+            builder.Services.AddEndpointsApiExplorer();
+            builder.Services.AddSwaggerGen();
+
+            var app = builder.Build();
+
+            // Configure the HTTP request pipeline.
+            if (app.Environment.IsDevelopment())
+            {
+                app.UseSwagger();
+                app.UseSwaggerUI();
+            }
+
+            app.UseHttpsRedirection();
+
+            app.UseAuthorization();
+
+            app.UseSession();
+
+            var summaries = new[]
+            {
+                "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
+            };
+
+            app.MapGet("/weatherforecast", (HttpContext httpContext) =>
+            {
+                // First check if there is a cached forecast stored in the session state
+                if (httpContext.Session.TryGetValue(ForecastCacheKey, out var serializedForecast))
+                {
+                    // If there is, deserialize it and return it
+                    var cachedForecast = JsonSerializer.Deserialize<WeatherForecast>(serializedForecast);
+
+                    if (cachedForecast != null)
+                    {
+                        return cachedForecast;
+                    }
+                }
+
+                // Otherwise if we made it here, generate a new forecast
+                var forecast = new WeatherForecast
+                {
+                    Date = DateTime.Now,
+                    TemperatureC = Random.Shared.Next(-20, 55),
+                    Summary = summaries[Random.Shared.Next(summaries.Length)]
+                };
+
+                // Save the new forecast in the session state and then return it
+                httpContext.Session.Set(ForecastCacheKey, JsonSerializer.SerializeToUtf8Bytes(forecast));
```

Comment:
nit: You might want to show how to put on an expiration on the cache. Works with the weather example since weather does change just not so fast that we need to "recompute" it for every request.

Reply:
Added an explanation above `options.IdleTimeout` in the setup.